File: TransCoder_tokenized_test_set_functions/python/PRUFER_CODE_TREE_CREATION.py
def printTreeEdges ( prufer, m ) :
    vertices = m + 2
    vertex_set = [ 0 ] * vertices
    for i in range ( vertices - 2 ) :
        vertex_set [ prufer [ i ] - 1 ] += 1
    print ( "The edge set E(G) is :" )
    j = 0
    for i in range ( vertices - 2 ) :
        for j in range ( vertices ) :
            if ( vertex_set [ j ] == 0 ) :
                vertex_set [ j ] = - 1
                print ( "(", ( j + 1 ), ", ", prufer [ i ], ") ", sep = "", end = "" )
                vertex_set [ prufer [ i ] - 1 ] -= 1
                break
    j = 0
    for i in range ( vertices ) :
        if ( vertex_set [ i ] == 0 and j == 0 ) :
            print ( "(", ( i + 1 ), ", ", sep = "", end = "" )
            j += 1
        elif ( vertex_set [ i ] == 0 and j == 1 ) :
            print ( ( i + 1 ), ")" )
    

File: TransCoder_tokenized_test_set_functions/java/PRUFER_CODE_TREE_CREATION.java
static void printTreeEdges ( int prufer [ ], int m ) {
  int vertices = m + 2;
  int vertex_set [ ] = new int [ vertices ];
  for ( int i = 0;
  i < vertices;
  i ++ ) vertex_set [ i ] = 0;
  for ( int i = 0;
  i < vertices - 2;
  i ++ ) vertex_set [ prufer [ i ] - 1 ] += 1;
  System . out . print ( "\nThe edge set E(G) is :\n" );
  int j = 0;
  for ( int i = 0;
  i < vertices - 2;
  i ++ ) {
    for ( j = 0;
    j < vertices;
    j ++ ) {
      if ( vertex_set [ j ] == 0 ) {
        vertex_set [ j ] = - 1;
        System . out . print ( "(" + ( j + 1 ) + ", " + prufer [ i ] + ") " );
        vertex_set [ prufer [ i ] - 1 ] --;
        break;
      }
    }
  }
  j = 0;
  for ( int i = 0;
  i < vertices;
  i ++ ) {
    if ( vertex_set [ i ] == 0 && j == 0 ) {
      System . out . print ( "(" + ( i + 1 ) + ", " );
      j ++;
    }
    else if ( vertex_set [ i ] == 0 && j == 1 ) System . out . print ( ( i + 1 ) + ")\n" );
  }
}